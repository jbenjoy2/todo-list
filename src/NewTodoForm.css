.NewTodoForm-btn {
	margin-top: 10px;
}
