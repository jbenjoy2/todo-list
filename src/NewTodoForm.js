import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import './NewTodoForm.css';

const NewTodoForm = ({ add }) => {
	const INITIAL_STATE = {
		task : ''
	};
	const [ formData, setFormData ] = useState(INITIAL_STATE);

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({
			...formData,
			[name] : value
		}));
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		add({ ...formData, id: uuid() });
		setFormData(INITIAL_STATE);
	};

	return (
		<div>
			<form onSubmit={handleSubmit}>
				<div>
					<label htmlFor="task">New Task</label>
					<input
						id="task"
						type="text"
						name="task"
						placeholder="Task"
						value={formData.task}
						onChange={handleChange}
					/>
				</div>
				<button className="NewTodoForm-btn">Add Task</button>
			</form>
		</div>
	);
};

export default NewTodoForm;
