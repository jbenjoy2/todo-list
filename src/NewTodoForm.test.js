import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

// smoke test
it('renders without crashing', () => {
	render(<NewTodoForm />);
});

// snapshot test
it('matches snapshot', () => {
	const { asFragment } = render(<NewTodoForm />);
	expect(asFragment()).toMatchSnapshot();
});

it('should call add function on form submit', () => {
	const mockAddFunc = jest.fn();

	const { getByText } = render(<NewTodoForm add={mockAddFunc} />);
	// grab submit button and click it
	const addBtn = getByText('Add Task');
	fireEvent.click(addBtn);

	// that add function should be called by submit handler
	expect(mockAddFunc).toHaveBeenCalled();
});
