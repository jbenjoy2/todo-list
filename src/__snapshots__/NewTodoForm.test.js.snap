// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div>
        <label
          for="task"
        >
          New Task
        </label>
        <input
          id="task"
          name="task"
          placeholder="Task"
          type="text"
          value=""
        />
      </div>
      <button
        class="NewTodoForm-btn"
      >
        Add Task
      </button>
    </form>
  </div>
</DocumentFragment>
`;
